src/Microsoft.Health.Fhir.Api/Features/Filters/ValidationModeFilterAttribute.cs

```diff
@@ -0,0 +1,75 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using EnsureThat;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Health.Fhir.Core.Features;
+using Microsoft.Health.Fhir.Core.Features.Operations;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.Core.Features.Validation;
+using Microsoft.Health.Fhir.Core.Models;
+
+namespace Microsoft.Health.Fhir.Api.Features.Filters
+{
+    [AttributeUsage(AttributeTargets.Class)]
+    internal class ValidationModeFilterAttribute : ActionFilterAttribute
+    {
+        public override void OnActionExecuting(ActionExecutingContext context)
+        {
+            EnsureArg.IsNotNull(context, nameof(context));
+
+            StringValues modes;
+            if (context.HttpContext.Request.Query.TryGetValue(KnownQueryParameterNames.Mode, out modes))
+            {
+                if (modes.Count > 1)
+                {
+                    throw new BadRequestException(Resources.MultipleModesProvided);
+                }
+
+                var idMode = context.ActionDescriptor.DisplayName.Contains("ValidateById", StringComparison.OrdinalIgnoreCase);
+                var mode = modes[0];
+                ParseMode(mode, idMode);
+            }
+        }
+
+        public static void ParseMode(string mode, bool idMode)
```

Comment:
nit: Instead of exposing this (and the other methods `ValidateId` and `ValidateType` ) as `public static` would it be cleaner to consolidate all these into a helper class (something like `ValidationHelpers`) since they are all related to validation?

Reply:
I refactored the controller to no longer need access to these methods, so I made them private.